include/deal.II/grid/grid_tools.h

```diff
@@ -1285,6 +1285,269 @@ namespace GridTools
       const bool                                             perform_handshake,
       const bool enforce_unique_mapping = false);
 
+
+    /**
+     * A function that fills DistributedComputeIntersectionLocationsInternal.
+     * @p intersection_requests are vertices of cells which describe the
+     * entities we want to search intersections for.
+     * The template parameter @p request_dim provides the dimension of the
+     * requesting intersection, e.g., if the intersection request corresponds
+     * to faces request_dim = dim-1 and for cells request_dim = dim.
+     */
+    template <int request_dim,
+              int dim,
+              int spacedim,
+              typename IntersectionRequest>
```

Comment:
Could you be more explicit about the type. `std::vector`, `std::array`? I guess `std::array` is okay if one has only one type of cells and one needs `std::vector` for mixed meshes!?

Reply:
Same discussion in https://github.com/dealii/dealii/pull/15165